providers/core.tcpsocket.js

```diff
@@ -262,38 +286,52 @@ Socket_chrome.errorStringOfCode = function(code) {
       'UNKNOWN';
 };
 
+/**
+ * Get the Freedom error code associated with a Chrome error string
+ * @method freedomErrorCode
+ * @static
+ * @private
+ * @param {String=} errorString The Chrome error string
+ * @return {String} The corresponding Freedom error string (often identical).
+ */
+Socket_chrome.freedomErrorCode = function(errorString) {
+  return Socket_chrome.FREEDOM_ERROR_MAP[errorString] || 'UNKNOWN';
+};
+
 /*
  * Alert a consumer that a socket is disconnected, with appropriate
  * error message.
  * @method dispatchDisconnect
  * @private
- * @param {Number} code the code returned by chrome when the socket closed.
+ * @param {Number=} code the code returned by chrome when the socket closed.
+ * @return {{errorcode: string, message: string}} Error JSON object.
  */
 Socket_chrome.prototype.dispatchDisconnect = function (code) {
-  if (!this.id) {
-    // Don't send more than one dispatchDisconnect event.
-    return;
+  var errorString = Socket_chrome.errorStringOfCode(code);
+  var freedomErrorCode = Socket_chrome.freedomErrorCode(errorString);
+  /** @type {string} */ var errorMessage;
+  if (freedomErrorCode === 'SUCCESS') {
+    errorMessage = 'Socket closed by call to close';
+  } else if(freedomErrorCode === 'CONNECTION_CLOSED') {
+    errorMessage = 'Connection closed gracefully';
+  } else {
+    errorMessage = 'Socket Error: ' + code + ': ' + errorString;
   }
 
-  Socket_chrome.removeActive(this.id);
-  delete this.id;
+  var errorObject = {
+    'errorcode': freedomErrorCode,
```

Comment:
Typo?

It's preventing the `errcode` field from getting through:
https://github.com/freedomjs/freedom-for-chrome/pull/62